zerver/views/auth.py

```diff
@@ -457,11 +457,13 @@ def start_social_login(request: HttpRequest, backend: str, extra_arg: Optional[s
         extra_url_params = {'idp': extra_arg}
     backends = ["github", "google", "gitlab"]
     # TODO: Add AzureAD also.
-    for backend in backends:
-        key_setting = "SOCIAL_AUTH_" + backend.upper() + "_KEY"
-        secret_setting = "SOCIAL_AUTH_" + backend.upper() + "_SECRET"
+    for backend_to_check in backends:
+        if not backend == backend_to_check:
+            continue
+        key_setting = "SOCIAL_AUTH_" + backend_to_check.upper() + "_KEY"
+        secret_setting = "SOCIAL_AUTH_" + backend_to_check.upper() + "_SECRET"
         if not (getattr(settings, key_setting) and getattr(settings, secret_setting)):
-            return redirect_to_config_error(backend)
+            return redirect_to_config_error(backend_to_check)
```

Comment:
wouldn't this be better to write as
```
if backend in backends:
        key_setting = "SOCIAL_AUTH_" + backend.upper() + "_KEY"
        secret_setting = "SOCIAL_AUTH_" + backend.upper() + "_SECRET"
        if not (getattr(settings, key_setting) and getattr(settings, secret_setting)):
            return redirect_to_config_error(backend)
```

Reply:
Ah, that's much better, I changed it, thanks!